### ISSUE TITLE
Supporting `esbuild-plugin-pipe`

### ISSUE THREAD

**Issue by AndrewLeedham** [unjs/unplugin#126]
Hi 👋, I have been loving using `unplugin` mostly for `vite` and `esbuild` plugins in parallel.

One snag that took a while to get my head around is using multiple `esbuild` plugins to transform some code, this works in piped bundlers like `rollup`, `vite` etc. but from what I understand not in `esbuild` as it exposes `onLoad` internally which runs once per file, and `unplugin` uses that when using `transform` meaning any subsequent plugins won't do anything.

A potential solution seems to be [`esbuild-plugin-pipe`](https://github.com/nativew/esbuild-plugin-pipe) which adds support for compatible plugins to pipe from one to another, passing along the output. Would support for this  be considered as part of `unplugin`?

**Comment by sxzz** [comment]
The esbuild-plugin-pipe repository is now archived. The unplugin library has an internal `onTransform` hook, allowing all unplugin plugins to be chained together on the user side. See https://github.com/vue-macros/vue-macros/blob/5fd78c159fd0cd8162bc6d990a3eb0c516216ee0/tsup.config.ts#L57
